astropy/io/votable/converters.py

```diff
@@ -1413,13 +1424,13 @@ def table_column_to_votable_datatype(column):
        A dict containing 'datatype' and 'arraysize' keys that can be
        set on a VOTable FIELD element.
     """
+    votable_string_dtype = None
+    if column.info.meta is not None:
+        votable_string_dtype = column.info.meta.get('_votable_string_dtype')
     if column.dtype.char == 'O':
-        if isinstance(column[0], bytes):
-            if _all_bytes(column[1:]):
-                return {'datatype': 'char', 'arraysize': '*'}
-        elif isinstance(column[0], str):
-            if _all_unicode(column[1:]):
-                return {'datatype': 'unicodeChar', 'arraysize': '*'}
+        if votable_string_dtype is not None:
+            string_dtype = votable_string_dtype
+            return {'datatype': string_dtype, 'arraysize': '*'}
```

Comment:
Why is `string_dtype = votable_string_dtype` needed? Maybe we can do the following instead?
```suggestion
            return {'datatype': votable_string_dtype, 'arraysize': '*'}
```